server/supernova/server/main.cpp

```diff
@@ -155,46 +157,45 @@ void start_audio_backend(server_arguments const& args) {
     int output_channels = args.output_channels;
 
     std::string input_device, output_device;
-    if (args.hw_name.empty()) {
-        boost::tie(input_device, output_device) = instance->default_device_names();
-    } else if (args.hw_name.size() == 1) {
-        input_device = output_device = args.hw_name[0];
-    } else {
-        input_device = args.hw_name[0];
-        output_device = args.hw_name[1];
+    if (!args.hw_name.empty()) {
+        if (args.hw_name.size() == 1) {
+            input_device = output_device = args.hw_name[0];
+        } else {
+            input_device = args.hw_name[0];
+            output_device = args.hw_name[1];
+        }
     }
 
-    cout << "opening portaudio device name: ";
-    cout << input_device << " / " << output_device << endl;
-
-    if (input_device == "nil") {
+    if (input_channels == 0)
         input_device.clear();
-        input_channels = 0;
-    }
-
-    if (output_device == "nil") {
+    if (output_channels == 0)
         output_device.clear();
-        output_channels = 0;
-    }
-
-    cout << "opening portaudio device name: ";
-    cout << input_device << " / " << output_device << endl;
 
+    std::cout << "requested audio devices:\n";
+    std::cout << "  In: " << input_device << "\n";
+    std::cout << "  Out: " << output_device << std::endl;
```

Comment:
consider improving this message if you have time, right now if i don't select hardware it just says:

```
requested audio devices:
  In: 
  Out: 
```

which almost looks like it might be an error or cause for concern

Reply:
Again, this is currently similar to scsynth + portaudio.
I can see how this is confusing. I have 3 ideas:
- `  In: (empty)` - consistent with the state of the string, but probably not meaningful
- `  In: (none)` - consistent with command line option (no device specified), but IMO would be confused with "no input"
- `  In: (default)` - consistent with the functionality - that would probably be my preference

Also note the funny part with specifying a single device (input OR output) - the other one becomes `"nil"`:
```supercollider
s.options.outDevice_("Built-in");
s.reboot;
```
```
...
Requested audio devices:
  In: nil
  Out: Built-in

Selecting default Core Audio input device

Opening audio devices:
  In: BlackHole
  Out: Built-in Output
```